guardian/media-atom-maker
# Log to Fluent Bit via stdout instead of directly to Kinesis

## What does this change?

This PR changes our logging setup. We no longer log directly to Kinesis, instead; Fluent Bit will pick up logs from `stdout` and send them to Kinesis. As part of this:
- `.conf` files for Fluent Bit are including in our build by `build.sbt`.
- We include new logging appenders in our `logback.xml` file.
- We remove the existing `KinesisLogging`
- (I've also specified a more specific long term support Node version.)

This PR is accompanied by a related PR in `editorial-tools-platform`.

As a side effect of this change, we now log slightly more, as our logs will include everything from `stdout` including logs that take place on the instance during build and deploy.


## How to test

1. Use the `media-atom-maker.yaml` file in our [ed-tools-platform PR](https://github.com/guardian/editorial-tools-platform/pull/578) to update the media-atom-maker-CODE stack in CloudFormation (`media-service` account). (I've already done this).
2. Ensure the AMI used for the boxes includes FluentBit. (I've done this too)
3. Deploy to CODE the latest build for this PR. (At time of writing, this has been done)
4. Check [logs.gutools.co.uk for media-atom-maker logs](https://logs.gutools.co.uk/s/editorial-tools/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-18h,to:now))&_a=(columns:!(message),filters:!(('$state':(store:appState),exists:(field:app),meta:(alias:!n,disabled:!f,index:'67604290-baa6-11e9-bea2-633437abb232',key:app,negate:!f,type:exists,value:exists)),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:'67604290-baa6-11e9-bea2-633437abb232',key:app.keyword,negate:!f,params:(query:media-atom-maker),type:phrase),query:(match_phrase:(app.keyword:media-atom-maker))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:'67604290-baa6-11e9-bea2-633437abb232',key:stage.keyword,negate:!f,params:(query:CODE),type:phrase),query:(match_phrase:(stage.keyword:CODE)))),index:'67604290-baa6-11e9-bea2-633437abb232',interval:auto,query:(language:kuery,query:''),sort:!())) while making changes on [CODE Media Atom Maker](https://video.code.dev-gutools.co.uk/). Any log doc with an `ami-id` field has arrived via fluentbit.

## Further info:

Along the way I logged via both Fluent Bit and the existing logger. All logs were duplicated, other than new logs that only came via Fluent Bit. This indicates that Fluent Bit is passing on all logs as intended
![Screenshot 2021-11-25 at 14 10 52](https://user-images.githubusercontent.com/34686302/143560671-b97542a6-fc83-4183-b30b-cdf7c54a8c28.png)
:
